scarb/src/compiler/db.rs

```diff
@@ -145,66 +145,25 @@ fn build_project_config(unit: &CairoCompilationUnit) -> Result<ProjectConfig> {
         .map(|component| {
             let experimental_features = component.package.manifest.experimental_features.clone();
             let experimental_features = experimental_features.unwrap_or_default();
-            // Those are direct dependencies of the component.
-            let dependencies_summary: Vec<&ManifestDependency> = component
-                .package
-                .manifest
-                .summary
-                .full_dependencies()
-                .collect();
 
-            // We iterate over all of the compilation unit components to get dependency's version.
-            let mut dependencies: BTreeMap<String, DependencySettings> = unit
-                .components
+            let dependencies: BTreeMap<String, DependencySettings> = component
+                .dependencies
                 .iter()
-                .filter(|component_as_dependency| {
-                    dependencies_summary.iter().any(|dependency_summary| {
-                        dependency_summary.name == component_as_dependency.package.id.name
-                    }) ||
-                        // This is a hacky way of accommodating integration test components,
-                        // which need to depend on the tested package.
-                        component_as_dependency
-                        .package
-                        .manifest
-                        .targets
-                        .iter()
-                        .filter(|target| target.kind.is_test())
-                        .any(|target| {
-                            target.group_id.clone().unwrap_or(target.name.clone())
-                                == component.package.id.name.to_smol_str()
-                            && component_as_dependency.cairo_package_name() != component.cairo_package_name()
-                        })
-                })
-                .map(|compilation_unit_component| {
+                .map(|compilation_unit_component_id| {
+                    let compilation_unit_component = unit.components.iter().find(|component| component.id == *compilation_unit_component_id)
+                        .expect("Dependency of a component is guaranteed to exist in compilation unit components");
                     (
                         compilation_unit_component.package.id.name.to_string(),
                         DependencySettings {
                             discriminator: (compilation_unit_component.package.id.name.to_string()
                                 != *CORELIB_CRATE_NAME)
-                                .then_some(compilation_unit_component.package.id.version.clone())
-                                .map(|v| v.to_smolstr()),
+                                .then_some(compilation_unit_component.id.clone())
```

Comment:
I wonder if we should hide the corelib edge case in cu too.

Reply:
Definitely. I don't think we want all tools to think about this case

Reply:
To be fair it doesn't even matter from the compiler's point of view if the corelib discriminator is `None` or not as long as `core` is a dependency for each package (which it is in Scarb)

https://github.com/starkware-libs/cairo/blob/932767340c4b8a762140bf9eba305f437587ac1b/crates/cairo-lang-semantic/src/resolve/mod.rs#L1077

I agree though that we should keep it as `None` for consistency, will change it in metadata and Scarb's internal model

Reply:
Are you sure it does not matter? I know this has not been working without this `if` when we first introduced it. If we can omit this logic, it would be best to do so I think 🤔